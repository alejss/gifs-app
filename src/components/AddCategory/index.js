import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ seetCategories }) => {
    const [inputValue, setinputValue] = useState('')

    const handleChangeText = (evn) => {
        const text = evn.target.value
        setinputValue(text)
    }

    const handleSubmit = (evn) => {
        evn.preventDefault()
        if (inputValue.trim().length > 2) seetCategories(categories => [inputValue, ...categories])
        setinputValue('')
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                placeholder='Add categories...'
                value={inputValue}
                onChange={handleChangeText}
            />
        </form>
    )
}

AddCategory.propTypes = {
    seetCategories: PropTypes.func.isRequired,
}
