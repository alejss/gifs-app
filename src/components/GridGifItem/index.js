import React from 'react'
import './style.css'

export const GridGifItem = ({ id, title, url }) => {
    return (
        <div className="card animate__animated animate__swing">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
