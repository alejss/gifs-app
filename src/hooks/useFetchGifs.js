import { useState, useEffect } from 'react'
import { getGifs } from '../services/gifs'

export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        gifs: [],
        loading: true,
    })

    useEffect(() => {
        const fetchGifs = async () => {
            const gifs = await getGifs(category)
            setstate({ gifs, loading: false })
        };
        fetchGifs()

    }, [category, setstate])

    return state
}