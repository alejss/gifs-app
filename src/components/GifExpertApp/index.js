import React, { useState } from 'react'
import './style.css'

// Components
import { AddCategory } from '../AddCategory'

export const GifExpertApp = () => {
  const [categories, seetCategories] = useState(['Cats', 'Dogs', 'Random'])
  return (
    <>
      <h2>Gif Expert App</h2>
      <hr />
      <AddCategory seetCategories={seetCategories} />
      <ol>
        {
          categories.map((categorie) => (<li key={categorie}>{categorie}</li>))
        }
      </ol>
    </>
  )
}
