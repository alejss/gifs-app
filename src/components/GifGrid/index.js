import React, { useState, useEffect } from 'react'
import './style.css'
import { GridGifItem } from '../GridGifItem'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs()
    }, [])

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Cats&limit=10&api_key=${process.env.REACT_APP_GIPHY_KEY}`
        const resp = await fetch(url)
        const { data } = await resp.json()

        const gifs = data.map(({ id, title, images }) => ({ id, title, url: images.downsized_medium.url }))
        setImages(gifs)
    }

    getGifs()

    return (
        <>
            <h3 key={category}>{category}</h3>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GridGifItem
                            key={image.id}
                            {...image}
                        />)
                    )
                }
            </div>
        </>
    )
}
