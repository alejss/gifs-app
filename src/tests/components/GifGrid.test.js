import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing <GifGrid />', () => {

    const category = 'Dogs';

    test('Testing Snapshot Component', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        
    }); 

})

