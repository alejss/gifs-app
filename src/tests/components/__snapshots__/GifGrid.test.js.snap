// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGrid /> Testing Snapshot Component 1`] = `
<Fragment>
  <h3
    key="Dogs"
  >
    Dogs
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
